### PR TITLE
Update typescript template

### LINKED ISSUE #153: Update typescript template
- The rollup config for typescript multiplatform must match the one from js multiplatform
- In multiplatform structure, the tsconfig.json and types have to be in the top-level directory

### PR DESCRIPTION
**Pull request by panwrona**
- Update typescript template
- Change resources to assets directory
- Update documentation

Resolve #153

### CODE REVIEW

File: template_multiplatform_components/lumin/tsconfig.json [hunk 1]
```diff
@@ -1,7 +0,0 @@
-{
```
Comment (kpiascik) [inline template_multiplatform_components/lumin/tsconfig.json]: we don't need this tsconfig file in this folder?

File: template_overlay_typescript_components/lumin/package.json [hunk 2]
```diff
@@ -30,7 +29,7 @@
   "dependencies": {
     "eslint-plugin-react": "^7.15.1",
     "magic-script-components": "^2.0.2",
-    "magic-script-components-lumin": "^1.0.13",
+    "magic-script-components-lumin": "^1.0.11",
```
Comment (kpiascik) [inline template_overlay_typescript_components/lumin/package.json]: this shouldn't be changed to 11 but stay as 13

File: template_overlay_typescript_components/package.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+    "dependencies": {
+        "magic-script-components": "2.0.2"
```
Comment (kpiascik) [inline template_overlay_typescript_components/package.json]: This should be `"^2.0.2"`

File: template_multiplatform_components/lumin/tsconfig.json (hunk 1 above)
Comment (panwrona) [reply template_multiplatform_components/lumin/tsconfig.json]: We need this in template_overlay_typescript_components, template_multiplatform_components is by default without typescript support

File: template_overlay_typescript_components/lumin/package.json (hunk 2 above)
Comment (grozdanov) [reply template_overlay_typescript_components/lumin/package.json]: Actually, it needs to be `^1.0.14` :)

**Pull request merged**